src/main/java/test/project/bookshop/repository/book/BookRepository.java

```diff
@@ -1,12 +1,23 @@
 package test.project.bookshop.repository.book;
 
+import java.util.Optional;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 import test.project.bookshop.model.Book;
 
 public interface BookRepository extends JpaRepository<Book, Long>,
         JpaSpecificationExecutor<Book> {
     Page<Book> findAllByCategoriesId(Long categoryId, Pageable pageable);
+
+    @EntityGraph(attributePaths = {"categories"})
+    @Query("SELECT b FROM Book b")
```

Comment:
do you really need both of these annotations?

Reply:
Without @EntityGraph I receive a LazyInitializationException.
Without @Query I can't name the methods according to the previous reviewer's suggestions because I don't know how to autogenerate methods with such naming
https://github.com/Lessorg/BookShop/pull/12#discussion_r1871172174

Reply:
mentor probably forgot about Spring Data JPA query generation, either option is fine, but I'd prefer sticking to base version without adding Query